src/components/typography/text/text.js

```diff
@@ -33,14 +36,58 @@ const Headline = props => {
 
 Headline.displayName = 'TextHeadline';
 Headline.propTypes = {
-  elementType: PropTypes.oneOf(['h1', 'h2', 'h3']).isRequired,
+  as(props, propName, componentName, ...rest) {
```

Comment:
We can use `requiredIf` to achieve this a bit more dense. For reference look [here](https://github.com/commercetools/ui-kit/blob/c85074af2ce933a89d62791068f6066568f4db31/src/components/fields/time-field/time-field.js#L34). I like the idea however to only make the prop required if (and only if) elementType is not given (during the migration phase).